source/usage-examples/changeStream.txt

```diff
@@ -0,0 +1,66 @@
+=================
+Watch for Changes
+=================
+
+.. default-domain:: mongodb
+
+Open a Change Stream
+--------------------
+
+You can keep track of changes to data in MongoDB, such as changes to a
+collection, database, or deployment by opening a :manual:`Change
+Stream</changeStreams/>`. A ``Change Stream`` allows applications to
+watch for changes to data and react to those changes. You can open a
+``Change Stream`` by calling the :node-api:`watch()
+</Collection.html#watch>` method on a collection, database, or
+deployment. 
+
+The ``watch()`` method takes an optional pipeline as the first parameter
+that is an array of stages to filter the :manual:`change events
+</reference/change-events/>` output. Also, ``watch()`` can take an
+additional options object as the second parameter. Set the ``fullDocument``
+field of the additional options object to ``'updateLookup'`` to receive
+an event that contains the changes to the document as well as the entire
+altered document. Alternatively, you can set ``fullDocument`` to the
+string ``'default'`` to only receive the altered document, but not the
+changes to the document. 
+
+Process the Change Stream Events
+--------------------------------
+
+You can capture events from a change stream instance by using a listener
+function. Call the watch() command to get a Change Stream instance. Add
+your listener function by calling the `EventEmitter.on()
+<https://nodejs.org/api/events.html#events_emitter_on_eventname_listener>`_
+method on that instance. Set the value of the first parameter to the
+string ``'change'`` and add your :mdn:`callback function
+<Glossary/Callback_function>` as the second parameter.  The callback
+triggers when a change event is emitted, providing the next available
+document. You can specify logic in the callback to process the event
+document when it is received. 
+
+Call the :node-api:`close() </ChangeStream.html#close>` method on the
+change stream instance to stop processing change events. This method
+closes the change stream and frees resources.
+
+Example
+-------
+
+The following example opens a ``Change Stream`` on the ``movies``
+collection. We create a ``changeListener`` variable that is listener
+function to receive and print change events that occur. 
+
+To emit a change event, we want to perform a change to the collection, such as insertion
```

Comment:
Nitpick: "To do x, we want to..." sounds less decisive than "To do x, perform a change..."

Reply:
I'd like to keep it this way. I think there are some cases where we want to follow the following format:

"<to do x, we want to do y>" 
"<we can do y by performing some action>" 

to prevent the monotony and runny-sentence-ness of "<to do x, we can do y performing some action>"

Reply:
You're saying:
> to prevent the monotony and runny-sentence-ness of "<to do x, we can do y performing some action>"

I said:
> "To do x, we want to..." sounds less decisive than "To do x, perform a change..."

I don't think these are the same. That said, what do you think of what I suggested? I think "we want to" is extraneous because it does not provide any additional information.

Reply:
Suggestion that omit any mention of "we want":

Example
--------
**The following example opens a ``Change Stream`` on the ``movies`` collection. We create a listener function to receive and print change events that occur.**

**To emit a change event, we perform a change to the collection, such as insertion with ``insertOne()``. To prevent ``insertOne()`` from executing before the listener function can register, we create a timer that uses setTimeout to wait 1 second. After the insertion of the document, the listener should receive this event.**

**After that, we create a second timer to wait an additional second after the insertion of the document to close the ``Change Stream`` instance using ``changeStream.close()``.**